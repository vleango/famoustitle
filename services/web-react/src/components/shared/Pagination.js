import React from 'react';
import { forEach, range } from 'lodash';
import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';

export default (props) => {
  const {currentPage, totalPages} = props;
  if(currentPage > totalPages) {
    return;
  }

  const pageBoxes = 5;
  let selectedPagePosition = currentPage % pageBoxes;

  if(selectedPagePosition === 0) {
    selectedPagePosition = pageBoxes;
  }

  let paginationItems = [];
  forEach(range(1, pageBoxes + 1), (index) => {
    let pageNumber = currentPage - selectedPagePosition + index;
    paginationItems.push(
      <PaginationItem key={"pagination-item-" + index} active={ selectedPagePosition === index }>
        <PaginationLink href="#">
          { pageNumber }
        </PaginationLink>
      </PaginationItem>
    );
  });

  return (
    <Pagination>
      <PaginationItem disabled={currentPage === 1}>
        <PaginationLink previous href="#" />
      </PaginationItem>
      { paginationItems }
      <PaginationItem disabled={currentPage === totalPages}>
        <PaginationLink next href="#" />
      </PaginationItem>
    </Pagination>
  );
}
