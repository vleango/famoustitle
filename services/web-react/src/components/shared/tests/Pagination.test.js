import React from 'react';
import { shallow } from 'enzyme';

import Pagination from '../Pagination';

let wrapper;

describe('Shared', () => {
  describe('Pagination', () => {
    wrapper = shallow(<Pagination />);

    it('should correctly render Pagination', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('currentPage == 1', () => {
      beforeEach(() => {
        wrapper = shallow(<Pagination currentPage={1} totalPages={10} />);
      });
      it('should set the currentPage active', () => {
        expect(wrapper.childAt(1).html().includes('active')).toBe(true);
      });

      it('should set the previousArrow disabled', () => {
        expect(wrapper.childAt(0).html().includes('disabled')).toBe(true);
      });
    });

    describe('currentPage < totalPages', () => {
      beforeEach(() => {
        wrapper = shallow(<Pagination currentPage={7} totalPages={10} />);
      });
      it('should set the currentPage active', () => {
        expect(wrapper.childAt(2).html().includes('active')).toBe(true);
      });

      it('should not set the previousArrow disabled', () => {
        expect(wrapper.childAt(0).html().includes('disabled')).not.toBe(true);
      });

      it('should not set the nextArrow disabled', () => {
        expect(wrapper.childAt(wrapper.children().length - 1).html().includes('disabled')).not.toBe(true);
      });

    });

    describe('currentPage > totalPages', () => {
      beforeEach(() => {
        wrapper = shallow(<Pagination currentPage={20} totalPages={10} />);
      });
      it('should not render pagination', () => {
        expect(wrapper.html()).toBe(null);
      });
    });

    describe('currentPage == totalPages', () => {
      beforeEach(() => {
        wrapper = shallow(<Pagination currentPage={10} totalPages={10} />);
      });
      it('should set the currentPage active', () => {
        expect(wrapper.childAt(wrapper.children().length - 2).html().includes('active')).toBe(true);
      });

      it('should not set the previousArrow disabled', () => {
        expect(wrapper.childAt(0).html().includes('disabled')).not.toBe(true);
      });

      it('should set the nextArrow disabled', () => {
        expect(wrapper.childAt(wrapper.children().length - 1).html().includes('disabled')).toBe(true);
      });
    });

  });

});
