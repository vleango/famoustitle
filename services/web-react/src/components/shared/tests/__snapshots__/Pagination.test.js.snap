// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Pagination should correctly render Pagination 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PaginationItem
          disabled={false}
          tag="li"
>
          <PaginationLink
                    href="#"
                    previous={true}
                    tag="a"
          />
</PaginationItem>,
        Array [
          <PaginationItem
            active={false}
            tag="li"
>
            <PaginationLink
                        href="#"
                        tag="a"
            />
</PaginationItem>,
          <PaginationItem
            active={false}
            tag="li"
>
            <PaginationLink
                        href="#"
                        tag="a"
            />
</PaginationItem>,
          <PaginationItem
            active={false}
            tag="li"
>
            <PaginationLink
                        href="#"
                        tag="a"
            />
</PaginationItem>,
          <PaginationItem
            active={false}
            tag="li"
>
            <PaginationLink
                        href="#"
                        tag="a"
            />
</PaginationItem>,
          <PaginationItem
            active={false}
            tag="li"
>
            <PaginationLink
                        href="#"
                        tag="a"
            />
</PaginationItem>,
        ],
        <PaginationItem
          disabled={true}
          tag="li"
>
          <PaginationLink
                    href="#"
                    next={true}
                    tag="a"
          />
</PaginationItem>,
      ],
      "tag": "ul",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <PaginationLink
            href="#"
            previous={true}
            tag="a"
/>,
          "disabled": false,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "href": "#",
            "previous": true,
            "tag": "a",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "pagination-item-1",
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": <PaginationLink
            href="#"
            tag="a"
/>,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": NaN,
            "href": "#",
            "tag": "a",
          },
          "ref": null,
          "rendered": NaN,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "pagination-item-2",
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": <PaginationLink
            href="#"
            tag="a"
/>,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": NaN,
            "href": "#",
            "tag": "a",
          },
          "ref": null,
          "rendered": NaN,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "pagination-item-3",
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": <PaginationLink
            href="#"
            tag="a"
/>,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": NaN,
            "href": "#",
            "tag": "a",
          },
          "ref": null,
          "rendered": NaN,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "pagination-item-4",
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": <PaginationLink
            href="#"
            tag="a"
/>,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": NaN,
            "href": "#",
            "tag": "a",
          },
          "ref": null,
          "rendered": NaN,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "pagination-item-5",
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": <PaginationLink
            href="#"
            tag="a"
/>,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": NaN,
            "href": "#",
            "tag": "a",
          },
          "ref": null,
          "rendered": NaN,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <PaginationLink
            href="#"
            next={true}
            tag="a"
/>,
          "disabled": true,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "href": "#",
            "next": true,
            "tag": "a",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PaginationItem
            disabled={false}
            tag="li"
>
            <PaginationLink
                        href="#"
                        previous={true}
                        tag="a"
            />
</PaginationItem>,
          Array [
            <PaginationItem
              active={false}
              tag="li"
>
              <PaginationLink
                            href="#"
                            tag="a"
              />
</PaginationItem>,
            <PaginationItem
              active={false}
              tag="li"
>
              <PaginationLink
                            href="#"
                            tag="a"
              />
</PaginationItem>,
            <PaginationItem
              active={false}
              tag="li"
>
              <PaginationLink
                            href="#"
                            tag="a"
              />
</PaginationItem>,
            <PaginationItem
              active={false}
              tag="li"
>
              <PaginationLink
                            href="#"
                            tag="a"
              />
</PaginationItem>,
            <PaginationItem
              active={false}
              tag="li"
>
              <PaginationLink
                            href="#"
                            tag="a"
              />
</PaginationItem>,
          ],
          <PaginationItem
            disabled={true}
            tag="li"
>
            <PaginationLink
                        href="#"
                        next={true}
                        tag="a"
            />
</PaginationItem>,
        ],
        "tag": "ul",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <PaginationLink
              href="#"
              previous={true}
              tag="a"
/>,
            "disabled": false,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "href": "#",
              "previous": true,
              "tag": "a",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "pagination-item-1",
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": <PaginationLink
              href="#"
              tag="a"
/>,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": NaN,
              "href": "#",
              "tag": "a",
            },
            "ref": null,
            "rendered": NaN,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "pagination-item-2",
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": <PaginationLink
              href="#"
              tag="a"
/>,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": NaN,
              "href": "#",
              "tag": "a",
            },
            "ref": null,
            "rendered": NaN,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "pagination-item-3",
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": <PaginationLink
              href="#"
              tag="a"
/>,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": NaN,
              "href": "#",
              "tag": "a",
            },
            "ref": null,
            "rendered": NaN,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "pagination-item-4",
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": <PaginationLink
              href="#"
              tag="a"
/>,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": NaN,
              "href": "#",
              "tag": "a",
            },
            "ref": null,
            "rendered": NaN,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "pagination-item-5",
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": <PaginationLink
              href="#"
              tag="a"
/>,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": NaN,
              "href": "#",
              "tag": "a",
            },
            "ref": null,
            "rendered": NaN,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <PaginationLink
              href="#"
              next={true}
              tag="a"
/>,
            "disabled": true,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "href": "#",
              "next": true,
              "tag": "a",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
